Keith/change icon css

keitheous opened the pull request:
[OFFAPPS-1064] CSS changes on badges, overflowing elements and highlighting border! 😄 

## Description
We're changing three things here.
1) Update badge colors to reflect September 2018 standards. Checkout the other pull request in references regarding this.
2) Fix the text area bug (please refer to screenshot on the jira ticket).
3) Fix overflowing badge icon and counts. 

## Tasks
- [x] pass lint and test
- [x] read up on September 2018 standards and identify what to change - colors colors 🌈
- [x] research on flexbox - as suggested by Tony
- [x] research on what Olaf said about textarea and paddings - came up with a hacky band-aid solution

## Implementation notes
### Text Area
<details>
<summary>Before and After</summary>
<img width="263" alt="screen shot 2018-10-03 at 2 00 15 pm" src="https://user-images.githubusercontent.com/17760485/46394823-a6299980-c72d-11e8-9690-af93e5a6c4eb.png">
<img width="264" alt="screen shot 2018-10-03 at 1 58 08 pm" src="https://user-images.githubusercontent.com/17760485/46394847-b8a3d300-c72d-11e8-9bcd-3fa20e8b0e65.png">
</details>
<details>
<summary>How?</summary>
<img width="131" alt="screen shot 2018-10-03 at 1 58 21 pm" src="https://user-images.githubusercontent.com/17760485/46394958-19331000-c72e-11e8-9424-ffd45fa7f32f.png">
</details>
Through console, realised that perhaps the text areas are inheriting the width of the divs they currently resides in. Gave it a width of 98.5% and shifted the box to the right a tiny bit by 2px, so we can see the left and right highlights.

_UPDATE_: Instead of my initial approach, Tony suggested that we shift ~~(squeeze)~~ the entire body horizontally by 3pixels instead. That way, the labels and fields of the app still look aligned. This solution is consistent with the style guidelines. 
<details>
<summary>Updated changes</summary>
<img width="164" alt="screen shot 2018-10-04 at 1 52 04 pm" src="https://user-images.githubusercontent.com/17760485/46452006-ca907f00-c7dc-11e8-9d01-3d4ed6a4c6f5.png">
<img width="317" alt="screen shot 2018-10-04 at 1 53 14 pm" src="https://user-images.githubusercontent.com/17760485/46452094-412d7c80-c7dd-11e8-8fb9-918b7811d09e.png">
</details>


### Badge Colors
Referring to the changes made in this [pull request](https://github.com/zendesk/zendesk_console/pull/16451/files) (September 2018 standards - in reference). Copy paste the hex colors for the different icons

<details>
<summary>Before(left) and After(right)</summary>
<img width="228" alt="bade_colors_before" src="https://user-images.githubusercontent.com/17760485/46395126-b726da80-c72e-11e8-8a7c-5f66d1eca806.png">
VS
<img width="223" alt="badge_colors_after" src="https://user-images.githubusercontent.com/17760485/46395136-c0b04280-c72e-11e8-8284-54b5ab870862.png">
</details>

<details>
<summary>Interested in the code? Before(left), After (right)</summary>
<img width="191" alt="badge_css_before" src="https://user-images.githubusercontent.com/17760485/46395159-d291e580-c72e-11e8-8604-9989846788e7.png">
<img width="172" alt="badge_css_after" src="https://user-images.githubusercontent.com/17760485/46395164-da518a00-c72e-11e8-8913-a09fb82bb0e4.png">
</details>

### Overflowing Badge and Counts
Tony taught me about flexbox not too long ago (attached learning material on flexbox in references). Chris suggested that there should not be a fixed width. Hence I removed the width from `<li>` and introduced flex to `.counts`. 

<details>
<summary>Before(left), After (right)</summary>
<img width="266" alt="screen shot 2018-10-03 at 4 42 32 pm" src="https://user-images.githubusercontent.com/17760485/46395318-47fdb600-c72f-11e8-997d-46924b5c2bfd.png">
<img width="265" alt="screen shot 2018-10-03 at 4 40 48 pm" src="https://user-images.githubusercontent.com/17760485/46395320-492ee300-c72f-11e8-99fa-0d5859e269f7.png">
</details>

## References
[JIRA ticket](https://zendesk.atlassian.net/browse/OFFAPPS-1064)

[September 2018 standards](https://github.com/zendesk/zendesk_console/pull/16451) 
_UPDATE_: Mandy provided [Main Source](https://cl.ly/8289e2f71308) - attn: Olaf

[learning material on flex box](https://jsfiddle.net/zer00ne/ajtd0dLc)



## CCs
@zendesk/apps-migration


## Risks
* low - shifting pixels around 👈 👉 👆 👇

ocke commented on src/main.scss:
```diff
@@ -124,6 +124,8 @@ i.icon_circle_arrow_left {
 
       textarea {
         height: 72px;
+        width: 98.5%;
```
should this be relative? Maybe fixed it better because it's a know width size anyway `320px`. So the textarea can be `316px` then.

keitheous replied:
When you mean relative? Do you mean adding `position: relative;`?

I was thinking perhaps we can do 315px and 👉 2px

<details>
<summary>As such?</summary>
<img width="265" alt="screen shot 2018-10-03 at 6 03 21 pm" src="https://user-images.githubusercontent.com/17760485/46398252-f279d700-c737-11e8-96ab-92d835e39968.png">
</details>

ocke replied:
sorry, i meant a fixed width, not a percentage.

yu-tian113 commented on src/main.scss:
```diff
@@ -124,6 +124,8 @@ i.icon_circle_arrow_left {
 
       textarea {
         height: 72px;
+        width: 98.5%;
```
I remember Mandy asked to squeeze the whole app by `padding: 0 3px`, otherwise, elements not aligned.

keitheous replied:
Hmm.. perhaps i'll come sit with you tomorrow. I understand what you mean about alignment. But I'm thinking it is abit hard to see the highlights if it is 100% width.

<details>
<summary>Click me </summary>
<img width="251" alt="screen shot 2018-10-03 at 6 05 13 pm 2" src="https://user-images.githubusercontent.com/17760485/46398143-a464d380-c737-11e8-89b0-67a4ff086011.png">
</details>

Let's discuss this tomorrow.

sostopher commented:
@m-lai FYI.

keitheous commented:
After addressing the comments, here is what it currently looks like now, @m-lai =)
<img width="318" alt="screen shot 2018-10-04 at 10 04 18 am" src="https://user-images.githubusercontent.com/17760485/46445986-00257000-c7bd-11e8-84ab-50a51c35949a.png">

m-lai commented:
Hey @keitheous all the changes look great; just one thing - I believe the text color for the "Closed" badge should be grey-700 and not white, like so: https://cl.ly/8289e2f71308

ocke commented on src/main.scss:
```diff
@@ -5,6 +5,7 @@
 
 html, body {
   overflow: hidden;
+  padding: 0 2.75px;
```
.75 pixel?

keitheous replied:
We're shifting the entire body horizontally by ~~2.75~~ three pixels so that we get to see the highlights on the sides of the text box. That way, the labels and text fields are still aligned. 

I initially went with 2.75 or 2.25 because this number was all we need in order for the highlights to fully show up without getting ✂️ off =)

keitheous replied:
<img width="317" alt="screen shot 2018-10-04 at 1 53 14 pm" src="https://user-images.githubusercontent.com/17760485/46452143-70dc8480-c7dd-11e8-94c5-4f2ad7398673.png">

For the full description, kindly refer to the PR > Implementation Notes > Text Area> *updated changes*

sostopher commented on src/main.scss:
```diff
@@ -80,27 +80,27 @@ i.icon_circle_arrow_left {
   &.closed {
     background: #D8DCDE;
-    color: #FFFFFF;
+    color: #F8F9F9;
```
This should be `#49545C`, which is `Gray-700`

sostopher replied:
Could we also potentially add the comments in for the Garden colours for their reference? eg. `/* Gray-700 */` next to it :)

sostopher commented on src/main.scss:
```diff
@@ -80,27 +81,27 @@ i.icon_circle_arrow_left {
   &.closed {
     background: #D8DCDE;
-    color: #FFFFFF;
+    color: #F8F9F9;
```
This should be `#49545C`, which is `Gray-700`.

We should also add in some comments on the corresponding Garden colour references eg. `/* Gray-700 */` to denote. (Not all of these are Garden)

m-lai replied:
@sostopher btw the values in Garden are spelled `grey-100` with an 'e' (in case this causes confusion down the road)

sostopher replied:
Noted. Thanks. Something else the other PR has 😛

keitheous replied:
Okie dokie, I'll fix this.

keitheous commented:
hello @sostopher , I've addressed some of the comments =) Let me know what you think?

keitheous commented:
hey @m-lai , I've fixed it to 
```
  &.closed {
    background: #D8DCDE; /* Grey-300 */
    color: #49545C; /* Grey-700 */
```

Thank you for catching this. =)

token-cjg commented:
:+1:

zendesk-deploy commented:
This PR was deployed to Production. Reference: <a href='https://samson.zende.sk/projects/user_data_app/deploys/580446' target='_blank'>v2.0.10</a>
